zomes/likes/code/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+        // are properties, i.e. defined in and thus part of the DNA.
+        liked_membrane: |object: MembraneObject| {
+            let agent = object.agent_key();
+            // Get the seed agents that are defined in the DNA
+            let seeds : Vec<String> = hdk::properties("seeds");
+
+            // Seeds are liked by definition
+            if seeds.include(agent) {
+                true
+            } else {
+                 // I'm using a LikedNotarization entry to constrain
+                 // the transitive requirement of this membrane
+                 // (that only likes of liked ones are valid).
+                 // We don't want to run a depth-search on the like-graph
+                 // every time we have to check the membrane - instead
+                 // we use the trusted DHT as cache:
+                 if hdk::get_links(agent, "like_notarization").size() > 0 {
+                    true
+                } else {
+                    // Ok, no seed agent and no cached notarization yet.
+                    // We have to check ourselves:
+                    let likes = hdk::get_links(agent, "received_likes");
+                    let min_likes = hdk::properties("required_number_of_likes");
+                    // if the number of likes is less than need, don't bother..
+                    if likes.size() >= min_likes {
+                        // But if we have enough likes we need to check
+                        // how many of them are actually valid, i.e. coming
+                        // from a liked source
+                        let num_valid_likes = likes.iter()
+                            .filter(|like| {
+                                let like_source = hdk::get_links(like, "source")[0];
+                                hdk::check_membrane(like_source)
+                            })
+                            .collect::<Vec<Like>>()
+                            .size();
+
+                        // If that number is high enough, we have to creat the
+                        // notarization entry:
+                        if num_valid_likes > min_likes {
+                            hdk::commit("liked_notarization", json!({
+                                liked_agent_key: agent,
+                                date: now(),
+                            }))
+                            true
+                        } else {
+                            false
+                        }
+                    } else {
+                        false
+                    }
+                }
+            }
+        }
+    },
+
+    main (public) : {
+        like: |agent_key: String| {
+            hdk::commit("like", json!({
+                liked_agent_key: agent_key,
+            }))
+        }
+    }
+}
+
+// I'm also defining a macro for entry types.
+// This is supposed to spit out rust structs and the JSON
+// definition at the same time, similiar to capabilities!.
+entry_types!{
```

Comment:
How about renaming the macro to `zome_entry_types!` so its consistent with `zome_functions!` ?

Reply:
The `zome_functions!` macro has been renamed to `capabilities!`, so I think we should keep it simple.